libcextract/ElfCXX.cpp

```diff
@@ -115,6 +133,92 @@ ElfObject::~ElfObject(void)
   }
 }
 
+Elf *ElfObject::decompress_gz(std::string &parser_path)
+{
+  const size_t CHUNK = 16384;
+
+  unsigned have;
+  unsigned char in[CHUNK];
+  unsigned char out[CHUNK];
```

Comment:
There is no need for this `out` buffer. The `out` pointer can alias with the `dest` buffer by something like
```
unsigned char *out = dest + dest_current;
```

Reply:
This also saves one memcpy.

Reply:
But whatever, I doubt this is a bottleneck in clang-extract.

Reply:
I kept the code as is in this case.